### ISSUE TITLE
Roundrobin subscriptions on Windows aren't configured

### ISSUE THREAD

**Issue by bodgit** [sensu/sensu-puppet#820]
## Description of problem

- What did you do?

Applied the following on a Windows host:
```
::sensu::subscription { 'roundrobin:foo':
  ensure => present,
}
```
- What happened?

A zero-length file `C:\opt\sensu\conf.d\subscription_roundrobin` is created

- What did you expect to happen?

The subscription should be created, with a suitable filename.

- How can someone reproduce the problem?

It looks like the subscription name should be sanitised as while the `:` character is fine in filenames on !Windows, it has special meaning as a drive letter separator which likely explains why the filename is truncated.

## Platform and version information

- Your OS: Windows 2012r2
- Your version of Puppet: 4.10.0
- Your version of Sensu: 1.0.2
- Your version of this module: 2.35.0

**Comment by alvagante** [comment]
@ghoneycutt giving a look to this

**Comment by alvagante** [comment]
@bodgit can you confirm that https://github.com/sensu/sensu-puppet/pull/846 solves the issue?

**Comment by ghoneycutt** [comment]
Released in v2.43.0
